src/sentry/api/authentication.py

```diff
@@ -126,11 +127,12 @@ def authenticate(self, request):
 class TokenAuthentication(StandardAuthentication):
     token_name = b'bearer'
 
-    def authenticate_credentials(self, token):
+    def authenticate_credentials(self, request, token_str):
+        token = SystemToken.from_request(request, token_str)
```

Comment:
Expanding our existing token authentication is better than adding a new type of authentication.